mantle/platform/qemu.go

```diff
@@ -1194,30 +1199,47 @@ func (builder *QemuBuilder) setupIso() error {
 			return errors.Wrapf(err, "running coreos-installer iso embed")
 		}
 		builder.configInjected = true
+	}
 
-		if kargsSupported, err := coreosInstallerSupportsISOKargs(); err != nil {
-			return err
-		} else if kargsSupported {
-			allargs := fmt.Sprintf("console=%s %s", consoleKernelArgument[system.RpmArch()], builder.AppendKernelArguments)
-			instCmdKargs := exec.Command("coreos-installer", "iso", "kargs", "modify", "--append", allargs, isoEmbeddedPath)
-			var stderrb bytes.Buffer
-			instCmdKargs.Stderr = &stderrb
-			if err := instCmdKargs.Run(); err != nil {
-				// Don't make this a hard error if it's just for console; we
-				// may be operating on an old live ISO
-				if len(builder.AppendKernelArguments) > 0 {
-					return errors.Wrapf(err, "running `coreos-installer iso kargs modify`; old CoreOS ISO?")
-				}
-				stderr := stderrb.String()
-				plog.Warningf("running coreos-installer iso kargs modify: %v: %q", err, stderr)
-				plog.Warning("likely targeting an old CoreOS ISO; ignoring...")
+	if kargsSupported, err := coreosInstallerSupportsISOKargs(); err != nil {
+		return err
+	} else if kargsSupported {
+		allargs := fmt.Sprintf("console=%s %s", consoleKernelArgument[system.RpmArch()], builder.AppendKernelArguments)
+		instCmdKargs := exec.Command("coreos-installer", "iso", "kargs", "modify", "--append", allargs, isoEmbeddedPath)
+		var stderrb bytes.Buffer
+		instCmdKargs.Stderr = &stderrb
+		if err := instCmdKargs.Run(); err != nil {
+			// Don't make this a hard error if it's just for console; we
+			// may be operating on an old live ISO
+			if len(builder.AppendKernelArguments) > 0 {
+				return errors.Wrapf(err, "running `coreos-installer iso kargs modify`; old CoreOS ISO?")
 			}
-		} else if len(builder.AppendKernelArguments) > 0 {
-			return fmt.Errorf("coreos-installer does not support appending kernel args")
+			stderr := stderrb.String()
+			plog.Warningf("running coreos-installer iso kargs modify: %v: %q", err, stderr)
+			plog.Warning("likely targeting an old CoreOS ISO; ignoring...")
 		}
-		builder.iso.path = isoEmbeddedPath
+	} else if len(builder.AppendKernelArguments) > 0 {
+		return fmt.Errorf("coreos-installer does not support appending kernel args")
 	}
 
+	if builder.isoAsDisk {
+		f, err := os.OpenFile(isoEmbeddedPath, os.O_WRONLY, 0)
+		if err != nil {
+			return errors.Wrapf(err, "opening ISO image for writing")
+		}
+		defer f.Close()
+		// Invalidate Boot System Identifier in the El Torito Boot
+		// Record Volume Descriptor so the system must boot via the
+		// MBR or ESP.  If we don't do this, QEMU's UEFI firmware
+		// will boot via El Torito anyway, which doesn't match what
+		// a lot of UEFI firmware does.
```

Comment:
To clarify the firmware here is https://www.tianocore.org/ - in theory we could have a patched version which simulates what other hardware does, right?  But clearly not worth blocking on.

Reply:
Yeah, I looked into that briefly, but couldn't figure out the codebase.  :disappointed: